tests/internal/cluster/cluster.go

```diff
@@ -171,3 +179,130 @@ func ExecCmdWithStdout(
 
 	return outputMap, nil
 }
+
+// ExecCmdWithRetries executes a command on the provided client on each node matching nodeSelector,
+// retrying on internal errors
+// "retries" times with a "interval" duration between retries, and ignores the stdout.
+func ExecCmdWithRetries(client *clients.Settings, retries uint,
+	interval time.Duration, nodeSelector, command string) error {
+	glog.V(90).Infof("Executing command '%s' with %d retries and interval %v. Node Selector: %v",
+		command, retries, interval, nodeSelector)
+
+	for retry := range retries {
+		err := ExecCmd(client, nodeSelector, command)
+		if isErrorExecuting(err) {
+			glog.V(90).Infof("Error during command execution, retry %d (%d max): %w", retry+1, retries, err)
+
+			time.Sleep(interval)
+
+			continue
+		}
+
+		return err
+	}
+
+	return fmt.Errorf("ran out of %d retries executing command %s", retries, command)
+}
+
+// ExecCmdWithStdoutWithRetries executes a command on the provided client,
+// retrying on internal errors "retries" times with a "interval"
+// duration between retries, and returns the stdout for each node.
+func ExecCmdWithStdoutWithRetries(
+	client *clients.Settings, retries uint, interval time.Duration,
+	command string, options ...metav1.ListOptions) (map[string]string, error) {
+	glog.V(90).Infof("Executing command with stdout '%s' with %d retries and interval %v. Options: %v",
+		command, retries, interval, options)
+
+	for retry := range retries {
+		outputs, err := ExecCmdWithStdout(client, command, options...)
```

Comment:
It looks like wait PollImmediate* pattern. Please use wait functions instead of sleep

Reply:
Fixed